tests/test_config_noise.py

```diff
@@ -776,12 +776,13 @@ def test_whiten():
     np.testing.assert_almost_equal(im4b.array, im4a.array, decimal=5)
 
     # Repeat with gain != 1
-    config['image']['noise']['gain'] = 3.7
+    gain = 3.7
+    config['image']['noise']['gain'] = gain
     galsim.config.RemoveCurrent(config)
     im5a = im1a.copy()
-    rn = math.sqrt(25-cv1a * 3.7**2)
+    rn = math.sqrt((25/gain-cv1a) * gain**2)
     rng.reset(rng1.duplicate())
-    im5a.addNoise(galsim.CCDNoise(sky_level=25, read_noise=rn, gain=3.7, rng=rng))
+    im5a.addNoise(galsim.CCDNoise(sky_level=25, read_noise=rn, gain=gain, rng=rng))
     im5b, cv5b = galsim.config.BuildStamp(config)
```

Comment:
I suggest adding a test along the lines of
```
    # Check that this is (moderately) close to the actual noise in the image.
    np.testing.assert_allclose(cv5b, np.var(im5b.array - im1a.array), rtol=0.1)
```
This checks that the noise in the image actually is close to the return value.
Then I think the right expected value is now:
```
target_var = 25/3.7 + 5**2/3.7**2
np.testing.assert_almost_equal(cv5b, target_var, decimal=5)
```

Reply:
Actually, adding that check of the cv return value for all the noise types is probably a good idea.  I think the rest of them come out fine already, but it's worth checking the claim that this return value is the current noise variance in ADU as the doc string says.

Reply:
Or, actually, I guess the doc string doesn't specify ADU.  Might be worth adding that clarification somewhere.

Reply:
OK. I added all of the tests here. The return value was already tested each time. Now there is a test for the variance.